doc/reference/modules/transactions.xml

```diff
@@ -604,22 +604,22 @@ finally
                     The session will enlist with the first scope in which the session is used (or opened).
                     As of NHibernate v5.0, it will enlist its connection in the transaction regardless of
                     connection string <literal>Enlist</literal> setting. Prior to v5.0, it was relying on
-                    that setting being considered <literal>true</literal>. If the connection was supplied
-                    to the session at its opening, the session will not handle the connection enlistment:
-                    when supplying a connection to the session, it is your responsibility to handle its
-                    transaction enlistment.
+                    that setting being considered <literal>true</literal>, and on acquiring the connection
+                    within the scope.
                 </para>
                 <para>
-                    Sub-scopes are not supported. Using a session within a scope then within a sub-scope
-                    will cause undefined behavior.
+                    Sub-scopes are not supported. The session will be enlisted in the first scope within
+                    which it was used, until this scope is committed or rollback. If auto-enlistment is
+                    enabled on the connection and the session used on others scopes than the one in which
+                    it is currently enlisted, the connection may enlist in another scope, and the session
+                    will then fail to use it.
```

Comment:
Worst for NH4, it can not even notice it. Only this PR will always throw if trying to use a session enlisted in another transaction than its connection. I have not explained here for conciseness.